src/backend/base/langflow/services/database/models/message/model.py

```
@@ -62,7 +62,15 @@
                 if hasattr(file, "path") and hasattr(file, "url") and file.path:
                     session_id = message.session_id
                     if session_id:
-                        image_paths.append(f"{session_id}{file.path.split(str(session_id))[1]}")
+                        # Split only once, if session_id not found, fall back to the basename to avoid indexError.
+                        parts = file.path.split(str(session_id), 1)
+                        if len(parts) > 1:
+                            fname = parts[1]
+                        else:
+                            import os
+
+                            fname = os.path.basename(file.path)
```

Ruff (0.11.9)

`os.path.basename()` should be replaced by `Path.name`

(PTH119)